src/u2f.ml

```diff
@@ -197,28 +199,28 @@ let verify_sig pub ~signature data =
   | Ok () -> Ok ()
 
 let verify_reg_sig cert app client_data kh key signature =
-  let h s = Mirage_crypto.Hash.SHA256.digest (Cstruct.of_string s) in
+  let h s = Digestif.SHA256.(to_raw_string (digest_string s)) in
   let data =
-    Cstruct.concat [
-      Cstruct.create 1 ;
+    String.concat "" [
+      String.make 1 '\000' ;
       h app ;
       h client_data ;
       kh ;
-      Mirage_crypto_ec.P256.Dsa.pub_to_cstruct key
+      Mirage_crypto_ec.P256.Dsa.pub_to_octets key
     ]
   in
   verify_sig (X509.Certificate.public_key cert) ~signature data
 
 let verify_auth_sig key app presence counter client_data signature =
   let data =
-    let h s = Mirage_crypto.Hash.SHA256.digest (Cstruct.of_string s) in
+    let h s = Digestif.SHA256.(to_raw_string (digest_string s)) in
     let p_c =
-      let b = Cstruct.create 5 in
-      if presence then Cstruct.set_uint8 b 0 1;
-      Cstruct.BE.set_uint32 b 1 counter;
-      b
+      let b = Bytes.create 5 in
+      if presence then Bytes.set_uint8 b 0 1;
+      Bytes.set_int32_be b 1 counter;
+      Bytes.unsafe_to_string b
     in
-    Cstruct.concat [ h app ; p_c ; h client_data ]
+    String.concat "" [ h app ; p_c ; h client_data ]
   in
   verify_sig (`P256 key) ~signature data
```

Comment:
Here as well we could compute the hash in chunks to avoid a string concatenation...